code/modules/mob/living/simple_animal/guardian/guardian_abilities.dm

```diff
@@ -0,0 +1,733 @@
+		stando.clear_alert("charge")
+		chargealert = null
+		stando.Shoot(A)
+
+/datum/guardian_abilities/charge/life_act()
+	. = ..()
+	if(stando.ranged_cooldown <= world.time)
+		if(!chargealert)
+			chargealert = stando.throw_alert("charge", /obj/screen/alert/cancharge)
+	else
+		stando.clear_alert("charge")
+		chargealert = null
+
+/datum/guardian_abilities/charge/ranged_attack(atom/targeted_atom)
+	charging = 1
+	stando.throw_at(targeted_atom, stando.range, 1, stando, 0, callback = CALLBACK(stando, .proc/charging_end))
+
+
+/datum/guardian_abilities/charge/move_act()
+	if(charging)
+		new /obj/effect/overlay/temp/decoy/fading(stando.loc,stando)
+	. = ..()
+
+/datum/guardian_abilities/charge/impact_act(atom/A)
+	if(!charging)
+		return ..()
+
+	else if(A)
+		if(isliving(A) && A != user)
+			var/mob/living/L = A
+			var/blocked = 0
+			if(stando.hasmatchingsummoner(A)) //if the user matches don't hurt them
+				blocked = 1
+			if(ishuman(A))
+				var/mob/living/carbon/human/H = A
+				if(H.check_shields(90, "[stando.name]", stando, attack_type = THROWN_PROJECTILE_ATTACK))
+					blocked = 1
+			if(!blocked)
+				L.drop_all_held_items()
+				L.visible_message("<span class='danger'>[stando] slams into [L]!</span>", "<span class='userdanger'>[stando] slams into you!</span>")
+				L.apply_damage(20, BRUTE)
+				playsound(get_turf(L), 'sound/effects/meteorimpact.ogg', 100, 1)
+				shake_camera(L, 4, 3)
+				shake_camera(stando, 2, 3)
+
+		charging = 0
+
+/datum/guardian_abilities/charge/snapback_act()
+	if(!charging)
+		..()
+
+//protector ability
+/datum/guardian_abilities/protector/handle_stats()
+	stando.melee_damage_lower = 7
+	stando.melee_damage_upper = 7
+	stando.range = 7 //worse for it due to how it leashes
+	stando.damage_coeff -= list(BRUTE = 0.3, BURN = 0.3, TOX = 0.3, CLONE = 0.3, STAMINA = 0, OXY = 0.3)
+	stando.toggle_button_type = /obj/screen/guardian/ToggleMode
+
+/datum/guardian_abilities/protector/boom_act(severity)
+	if(severity == 1)
+		stando.adjustBruteLoss(400) //if in protector mode, will do 20 damage and not actually necessarily kill the user
+	else
+		..()
+	if(toggle)
+		stando.visible_message("<span class='danger'>The explosion glances off [stando]'s energy shielding!</span>")
+
+/datum/guardian_abilities/protector/adjusthealth_act(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(. > 0 && toggle)
+		var/image/I = new('icons/effects/effects.dmi', stando, "shield-flash", MOB_LAYER+0.01, dir = pick(cardinal))
+		if(stando.namedatum)
+			I.color = stando.namedatum.colour
+		flick_overlay_view(I, stando, 5)
+
+/datum/guardian_abilities/protector/handle_mode()
+	if(cooldown > world.time)
+		return 0
+	cooldown = world.time + 10
+	if(toggle)
+		stando.cut_overlays()
+		stando.melee_damage_lower = initial(stando.melee_damage_lower)
+		stando.melee_damage_upper = initial(stando.melee_damage_upper)
+		stando.speed = initial(stando.speed)
+		stando.damage_coeff = initial(stando.damage_coeff)
```

Comment:
Ha ha, oh no. No, this won't work; you're going to have to reset all values touched in handle_stats() and then reapply all the handle_stats() changes.
Also, this specifically won't work because you can't initial() lists; you have to set it manually.